### ISSUE TITLE
Manfiest Generation / Validation / PR User Experience Problems

### ISSUE THREAD

**Issue by chrpai** [microsoft/winget-pkgs#669]
# Description of the new feature/enhancement

Complete/Sync/Test business rules in CreateYAML.PS1, WinGet,  BOT and Documentation

A user would be able to succesfully submit a PR on the first attempt.  I've seen at least several inconsistencies in my first (failed) attempt.

The CreateYAML.PS1 accepted an arch (x86, x64) that winget validate failed.  The documentation says enumeration of supported architectures but does not say what these values are and how many you can use.

The BOT failed my ID saying that it must be Publisher.Package.  The CreateYAML.ps1 and winget validate and documentation doesn't enforce this.  In fact the web documentations good/better examples and winget search  show that this isn't a rule at all.

The BOT suggests that I should use VS Code when the GitHub repos suggests using CreateYAML.PS1.

These inconsistencies really drag down the UX of the people that you are trying to get to contribute to this repos. 

# Proposed technical implementation details (optional)

I'd consistency and great documentation so that any PR failures are clearly my fault and not secret handshake problems.

**Comment by chrpai** [comment]
Glad to know I'm not alone!
![image](https://user-images.githubusercontent.com/43178814/83167135-7931b380-a0d5-11ea-88ec-1740048a502c.png)

**Comment by denelon** [comment]
@chrpai we've made quite a few improvements over the last few months. Many of the issues have been related to a combination of business rules being implemented in multiple places. We've consolidated much of this by leveraging the client code in the pipelines validating manifests.

In addition, several of the steps in the validation process are inherently closed systems with limited output. Many changes have already been implemented to provide better messaging directly to users when they submit a PR and there are issues with invalid manifest files, SHA256 codes, SmartScreen validation, and other static analysis.

We have also been working with some of these other systems/internal teams to expose more granular output to us so we can in turn expose it to users directly in PR comments (and labels).

We currently don't support multiple architectures in one manifest, but that's on the backlog.
https://github.com/microsoft/winget-cli/issues/132
